gauntlet/packages/gauntlet-solana/src/commands/middlewares.ts

```diff
@@ -20,11 +22,21 @@ export const withProvider: Middleware = (c: SolanaCommand, next: Next) => {
   return next()
 }
 
-export const withWallet: Middleware = (c: SolanaCommand, next: Next) => {
+export const withWallet: Middleware = async (c: SolanaCommand, next: Next) => {
+  if (c.flags.withLedger || boolean(process.env.WITH_LEDGER)) {
+    logger.info('Loading Ledger wallet')
+    const path = c.flags.ledgerPath || DEFAULT_DERIVATION_PATH
+    c.wallet = await LedgerWallet.create(path)
+    console.info(`Operator address is ${c.wallet.publicKey}`)
+    return next()
```

Comment:
This could be a good place to disconnect the wallet after `next()` executes, and before this middleware returns. But not 100% sure if that is the workflow of how middlewares are executed...

Reply:
This is run just before the execution. Ideally we'd want it after it